src/test/java/com/answerdigital/answerking/controller/OrderControllerTest.java

```diff
@@ -0,0 +1,178 @@
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@AutoConfigureMockMvc(addFilters = false)
+@ExtendWith(SpringExtension.class)
+@WebMvcTest(OrderController.class)
+class OrderControllerTest {
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    OrderService orderService;
+
+    @MockBean
+    OrderRepository orderRepository;
+
+    private final OrderTestBuilder orderTestBuilder;
+
+    private final OrderRequestTestBuilder orderRequestTestBuilder;
+
+    private final LineItemResponseTestBuilder lineItemResponseTestBuilder;
+
+    private final OrderMapper orderMapper;
+
+    public OrderControllerTest() {
+        orderTestBuilder = new OrderTestBuilder();
+        orderRequestTestBuilder = new OrderRequestTestBuilder();
+        orderMapper = Mappers.getMapper(OrderMapper.class);
+        lineItemResponseTestBuilder = new LineItemResponseTestBuilder();
+    }
+
+    @Test
+    void getAllOrdersReturnListOfOrderObjects() throws Exception {
+        //given
+        final Order order = orderTestBuilder.withDefaultValues().build();
+        final OrderResponse orderResponse = orderMapper.orderToOrderResponse(order);
+        given(orderService.findAll()).willReturn(List.of(orderResponse));
+
+        //when
+        final RequestBuilder request = MockMvcRequestBuilders.get("/orders");
+
+        final MockHttpServletResponse response = mvc.perform(request).andReturn().getResponse();
+
+        //then
+        assertEquals(HttpStatus.OK.value(), response.getStatus());
+        assertFalse(response.getContentAsString().isEmpty());
+        final ObjectMapper mapper = new ObjectMapper();
+        final JsonNode jsonNodeResponse = mapper.readTree(response.getContentAsString());
+        assertEquals(orderResponse.getId(), jsonNodeResponse.get(0).get("id").asLong());
+        assertTrue(jsonNodeResponse.get(0).get("lineItems").isArray());
+        assertTrue(jsonNodeResponse.isArray());
```

Comment:
I think we should check jsonNodeResponse.isArray() before jsonNodeResponse.get(0).get("lineItems").isArray()

Reply:
good point, switched them around